low-level/sensors/impl/centos_7/disk_monitor.py

```diff
@@ -216,10 +221,20 @@ def run(self):
         try:
             # Connect to dbus system wide
             self._bus = SystemBus()
+            try:
+                # Obtain a disk manager interface for monitoring drives
+                disk_systemd = self._bus.get_object('org.freedesktop.UDisks2',
+                    '/org/freedesktop/UDisks2')
+                self._iem.iem_fault_resolved("UDISKS2_UNAVAILABLE", "UDISKS2_AVAILABLE")

```

Comment:
every iteration lot of code getting executed to know if resolved IEM to generate or  not. Instead track with a flag in memory, initialize that on generating fault IEM & first time on entering the sensor through this iem_fault_resolved(),to take care in case of node reboot or sspl restart

Reply:
Done, 
defined fault_iems list in iem.py file, and in case of fault, event_name will be added in this list, and then before raising fault_resolved we just check if that event is present or not in 'fault_iems' list.

In case of node reboot/sspl restart . sensor will call check_exsisting_fault_iems() function(defined in iem.py file) 
at the time of iitialization, and 'check_exsisting_fault_iems()' function will check for all fault events, if it is present in persistance cache that event gets added in 'fault_iems' list.